src/models/output.ts

```diff
@@ -0,0 +1,155 @@
+/**
+ * Copyright (c) Hathor Labs and its affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+import { OP_GREATERTHAN_TIMESTAMP, OP_DUP, OP_HASH160, OP_EQUALVERIFY, OP_CHECKSIG } from '../opcodes';
+import { MAX_OUTPUT_VALUE_32, MAX_OUTPUT_VALUE, TOKEN_AUTHORITY_MASK, TOKEN_MINT_MASK, TOKEN_MELT_MASK, TOKEN_INDEX_MASK } from '../constants';
+import { OutputValueError } from '../errors';
+import { util } from 'bitcore-lib';
+import helpers from '../utils/helpers';
+import Address from './address';
+
+type optionsType = {
+  tokenData?: number,
+  timelock?: number | null,
+};
+
+const defaultOptions = {
+  tokenData: 0,
+  timelock: null,
+};
+
+
+class Output {
+  // Output value as an integer
+  value: number;
+  // Address object of the value destination
+  address: Address;
+  // tokenData of the output
+  tokenData: number;
+  // Timestamp of the timelock of the output
+  timelock: number | null;
+
```

Comment:
Might be worth having a `tokenUid` field here (optional). From the perspective of a user, I want to create an outputs like `const output = new Output(value, address, tokenUid)`.

Later on, I create a transaction with it: `const tx = new Transaction(inputs, [output])`. And it also would be good not having to deal with the token data and tokens array manually. So something like this would set the tokens array and the corresponding token data for each output: `tx.finalizeTokenData()`.

A user that wants to send a tx could use it like this:
```
const o1 = new Output(value1, address1, tokenA);
const o2 = new Output(value2, address2, tokenB);
const o3 = new Output(value3, address3, tokenA);

const tx = new Transaction(inputs, [o1, o2, o3]);
tx.finalizeTokenData();
// still need to calculate weight, maybe finalize should do it all
send(tx);
```

Reply:
I'd rather keep only the original fields and add some methods like `setTokenUid(...)`, `getTokenUid(...)`, etc. We can add a `tokenUid` to the constructor as well.

I don't like to create a specific `tokenUid` field because it might conflict with the `tokenData`. It will probably be a misuse but we should prevent users from making mistakes.

Reply:
I like the idea of `tokenUid` to be honest, it's much more intuitive to create an output with this, instead of requiring the user to understand which `tokenData` to use. However it's not that simple to create a `finalizeTokenData` method, it would require us to add many other optional fields. The normal case if easy, HTR is `tokenData = 0` and each `tokenUid` you increment one but if the output is a mint or melt, you will never be able to set the correct `tokenData`. For that we need to have more optional fields in the output, e.g. `isMint` or `isMelt`.

I suggest we keep simple as it is right now and when we start using these models we can improve it the way we think is better.

Reply:
But there's no such thing as `tokenUid`. I agree that we should keep it simple for users, so we should have methods to do that properly (e.g., `setTokenUid(...)`, `getTokenUid(...)`).

Reply:
As we've discussed on a call we will keep the `Output` class as it is, with the `tokenData` field and discuss about how to make it easier for users to create txs/outputs. 

Issue created for further discussion: https://github.com/HathorNetwork/hathor-wallet-lib/issues/159